src/rpc/blockchain.cpp

```diff
@@ -2470,6 +2470,60 @@ static RPCHelpMan dumptxoutset()
     };
 }
 
+static RPCHelpMan getblocklocations()
+{
+    return RPCHelpMan{"getblocklocations",
+                "\nEXPERIMENTAL warning: this call may be removed or changed in future releases.\n"
+                "\nReturns a JSON for the location of 'blockhash' and its previous 'nblocks' block and undo data.\n",
+                {
+                    {"blockhash", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The block hash"},
+                    {"nblocks", RPCArg::Type::NUM, RPCArg::Optional::NO, "Number of block locations to return"},
+                },
+                {
+                    RPCResult{
+                        RPCResult::Type::ARR, "", "",
+                        {
+                            {RPCResult::Type::NUM, "file", "blk*.dat/rev*.dat file index"},
+                            {RPCResult::Type::NUM, "data", "block data file offset"},
+                            {RPCResult::Type::NUM, "undo", "undo data file offset (if exists)"},
+                            {RPCResult::Type::STR_HEX, "prev", "previous block hash"},
+                        }
+                    },
+                },
+                RPCExamples{
+                    HelpExampleCli("getblocklocation", "\"00000000c937983704a73af28acdec37b049d214adbda81d7e2a3dd146f6ed09\" 10")
+                },
+                [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue {
+
+    uint256 hash(ParseHashV(request.params[0], "blockhash"));
+    size_t nblocks = request.params[1].get_int();
+
+    const CBlockIndex* pblockindex = WITH_LOCK(cs_main, return LookupBlockIndex(hash));
+    if (!pblockindex) {
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Block not found");
```

Comment:
I suppose this is hit (also) when the block is pruned? Perhaps document it?

Reply:
Good point - added a check in https://github.com/bitcoin/bitcoin/pull/20702/commits/9d5c2bb85c2601ad54fa9df49b4f13e0490fb421#diff-decae4be02fb8a47ab4557fe74a9cb853bdfa3ec0fa1b515c0a1e5de91f4ad0bR2506.